src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -193,6 +191,15 @@ public function validate(mixed $entity, Constraint $constraint)
             ->addViolation();
     }
 
+    private function ignoreNullForField(UniqueEntity $constraint, string $fieldName): bool
+    {
+        if (\is_bool($constraint->ignoreNull)) {
+            return $constraint->ignoreNull;
```

Comment:
I'm not sure how safe it'd be, but typewise property `ignoreNull` should now become an array of strings in the constraint:

false => []
true => copy fields
string => [string]
array => ok

With this in place we'd be always dealing with `in_array` branch

Reply:
I was afraid it'll end being classified as a BC break, but worth trying anyway :)